python_modules/dagster-webserver/dagster_webserver/external_assets.py

```diff
@@ -114,15 +120,105 @@ async def handle_report_asset_materialization_request(
     return JSONResponse({})
 
 
+async def handle_report_asset_check_request(
+    context: BaseWorkspaceRequestContext,
+    request: Request,
+) -> JSONResponse:
+    # Record a runless asset check evaluation event.
+    # The asset key is passed as url path with / delimiting parts or as a query param.
+    # Properties can be passed as json post body or query params, with that order of precedence.
+
+    body_content_type = request.headers.get("content-type")
+    if body_content_type is None:
+        json_body = {}
+    elif body_content_type == "application/json":
+        json_body = await request.json()
+    else:
+        return JSONResponse(
+            {
+                "error": (
+                    f"Unhandled content type {body_content_type}, expect no body or"
+                    " application/json"
+                ),
+            },
+            status_code=400,
+        )
```

Comment:
seems like there is some code consolidation to be had with `handle_report_asset_materialization_request` as this is pretty much pure [![copypasta](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/WjlTemxb6oA4PgZFmj08/ad1d6316-1e6b-41d1-baa1-4e01688fae82/copypasta.png)](https://app.graphite.dev/macros/dagster-io)

Reply:
the remaining copy-pasta-esque code have control flow `return` statements, so factoring them is non trivial and i decided that the few lines of simple code duplication were preferable.